src/Data/Foldable.purs

```diff
@@ -225,6 +227,47 @@ intercalate sep xs = (foldl go { init: true, acc: mempty } xs).acc
   go { init: true } x = { init: false, acc: x }
   go { acc: acc }   x = { init: false, acc: acc <> sep <> x }
 
+-- | `foldMap` but with each element surrounded by some fixed value.
+-- |
+-- | For example:
+-- |
+-- | ```purescript
+-- | > surroundMap "*" show []
+-- | = "*"
+-- |
+-- | > surroundMap "*" show [1]
+-- | = "*1*"
+-- |
+-- | > surroundMap "*" show [1, 2]
+-- | = "*1*2*"
+-- |
+-- | > surroundMap "*" show [1, 2, 3]
+-- | = "*1*2*3*"
+-- | ```
+surroundMap :: forall f a m. (Foldable f, Semigroup m) => m -> (a -> m) -> f a -> m
+surroundMap d t f = unwrap (foldMap joined f) d
+  where joined a = Endo \m -> d <> t a <> m
```

Comment:
That would be a different function: requiring a `Monoid` instead of just a `Semigroup` - but I don't mind changing it to that if you prefer?  With `Endo` we effectively get to use the delimiter `d` as a substitute `mempty`.